acceptance/tests/custom_facts/custom_facts_aggregate_resolutions.rb

```diff
@@ -0,0 +1,90 @@
+test_name 'Facter should handle aggregated custom facts' do
+  tag 'risk:high'
+
+  fact_content1 = <<-EOM
+    Facter.add(:test_fact) do
+      has_weight 100000
+      setcode do
+        "test fact"
+      end
+    end
+
+    Facter.add(:test_fact, :type => :aggregate) do
+      has_weight 10000
+      chunk(:one) do
+        'aggregate'
+      end
+      
+      chunk(:two) do
+          'fact'
+      end
+
+      aggregate do |chunks|
+        result = ''
+      
+        chunks.each_value do |str|
+          result += str
+        end
+      
+        result
+      end
+    end
+  EOM
+
+  fact_content2 = <<-EOM
+  Facter.add(:test_fact) do
+    has_weight 10000
+    setcode do
+      "test fact"
+    end
+  end
+
+  Facter.add(:test_fact, :type => :aggregate) do
+    has_weight 100000
+    chunk(:one) do
+      'aggregate'
+    end
+    
+    chunk(:two) do
+        'fact'
+    end
+
+    aggregate do |chunks|
+      result = ''
+    
+      chunks.each_value do |str|
+        result += str
+      end
+    
+      result
+    end
+  end
+EOM
+
+  agents.each do |agent|
+    fact_dir1 = agent.tmpdir('fact1')
+    fact_file = File.join(fact_dir1, 'test_facts.rb')
+    create_remote_file(agent, fact_file, fact_content1)
+    
+    fact_dir2 = agent.tmpdir('fact2')
+    fact_file = File.join(fact_dir2, 'test_facts.rb')
+    create_remote_file(agent, fact_file, fact_content2)
+
+    teardown do
+      agent.rm_rf(fact_dir1)
+      agent.rm_rf(fact_dir2)
+    end
+
+    step "Agent: Verify test_fact from #{fact_file}" do
```

Comment:
In both messages you'll print the same fact_file name. You need different variables for them